### PR TITLE
#911 & # 912 refactor: Archive + hackathonDetails

### PR DESCRIPTION
**Pull request by reyhanrumengan**

### CODE REVIEW

File: hms-app/src/pages/Archive.tsx [hunk 1]
```diff
@@ -1,12 +1,77 @@
...
+import { NULL_DATE } from '../common/constants'
+import { getHackathonDetails } from '../actions/HackathonActions'
+import { useMsal } from '@azure/msal-react'
+import HackathonHeader from '../components/HackathonHeader'
+import IdeaCardList from '../components/lists/IdeaCardList'
+import { getIdeaDetails } from '../actions/IdeaActions'
 
 export default function Archive() {
+  const { instance } = useMsal()
+  const [hackathonData, setHackathonData] = useState({
```
Comment (verzano) [inline hms-app/src/pages/Archive.tsx:15]: why not done in a similar way as in line 25 like so?
```
const [hackathonData, setHackathonData] = useState<Hackathon>()
```

File: hms-app/src/pages/Archive.tsx [hunk 2]
```diff
@@ -24,12 +89,21 @@ export default function Archive() {
...
-      {selectedHackathonId && (
-        <HackathonDetails
-          hackathonId={selectedHackathonId}
-          type={HackathonDetailsType.Archive}
-        />
-      )}
+      {hackathonData.startDate !== NULL_DATE &&
+        hackathonData.startDate?.toString() !== 'Invalid Date' &&
+        !isHackathonLoading &&
+        !isHackathonError && (
```
Comment (verzano) [inline hms-app/src/pages/Archive.tsx:95]: this could probably be extracted to a private function with a name that better explains what it is checking

Comment (verzano) [reply hms-app/src/pages/Archive.tsx:95]: `hackathonData.startDate?.toString() !== 'Invalid Date'` this bit also feels a bit odd to me 🤔

Comment (reyhanrumengan) [reply hms-app/src/pages/Archive.tsx:95]: about the 'Invalid Date', I'm also not so sure, it was from the HackathonDetails. 
And `hackathonData.startDate?.toString() !== 'Invalid Date'` is also found in a lot of other places (other components).

Do we actually not need it here?

**Comment by rz-p** [comment]
Just a quick question: the hackathon will only be archived automatically **after** the end date has passed? Are there any way for the admin to archive manually?

@LetsPoke @verzano

**Comment by LetsPoke** [comment]
> Just a quick question: the hackathon will only be archived automatically **after** the end date has passed? Are there any way for the admin to archive manually?
> 
> @LetsPoke @verzano

the archive is only ment to see old hackathons, so there is no need to manually archive a hackathon (for now)

File: hms-app/src/pages/Archive.tsx [hunk 3]
```diff
@@ -1,12 +1,82 @@
...
+          })
+          .includes(ideaData.id)
+      ) {
+        setRelevantIdeaList((relevantIdeaList) => {
+          return [...relevantIdeaList, ideaData]
+        })
+      }
+  }, [ideaData])
+
+function validHackhaton() {
```
Comment (LetsPoke) [inline hms-app/src/pages/Archive.tsx:71]: typo

**Review by LetsPoke — CHANGES_REQUESTED** [review]
looks good (except typo)

but same with prettier, is it possible that "indent" is set to 4?

**Comment by reyhanrumengan** [comment]
> 

Which part you mean by this? Is it the .prettierrc or the .eslintrc.json?

**Comment by reyhanrumengan** [comment]
> looks good (except typo)
> 
> but same with prettier, is it possible that "indent" is set to 4?


Which part you mean by this? Is it the .prettierrc or the .eslintrc.json?

**Comment by LetsPoke** [comment]
> > looks good (except typo)
> > but same with prettier, is it possible that "indent" is set to 4?
> 
> Which part you mean by this? Is it the .prettierrc or the .eslintrc.json?

not 100% sure probably both if lint doesnt complain

**Pull request merged**